e2e/nomostest/wait.go

```diff
@@ -75,6 +75,18 @@ func Wait(t testing.NTB, opName string, timeout time.Duration, condition func()
 	t.Logf("took %v to wait for %s", took, opName)
 }
 
+// WaitFor waits for the specified object to meet the specified predicates.
+// Predicates must accept the object as Unstructured.
+func WaitFor(nt *NT, gvk schema.GroupVersionKind, name, namespace string, predicates []Predicate, opts ...WaitOption) {
```

Comment:
Maybe something a bit more explicit like `WaitForObject` or `WaitForValidate` - WDYT?

Reply:
It waits for the predicates passed in, and predicates are generic conditions. The only reason this is different from Wait() is that it does the object GET for you and passes it to the function so that the functions don't need to do the GET. 

But I can call it WaitForPredicates if you prefer.

Reply:
Or WaitForObject?

Reply:
WaitForObject would be great if the predicates were optional. But I dont know how to make them optional and also have optional wait options...

Reply:
You could potentially nest the function, e.g.
```golang
func WaitForObject(nt *NT, gvk schema.GroupVersionKind, name, namespace string, predicates ...Predicate) func(opts ...WaitOption) {
    return func(opts ...WaitOption) {
    ...
    }
}
```
Although maybe that's overkill. I'm also good with `WaitForPredicates`

Reply:
This is one of those times I wish a varargs could accept either one type or another type, but not any type. For now, I made it take any, which makes it a runtime problem, but avoids the problem of needing two optional arrays.